### PR TITLE
Small tidying (depends on #49)

### PR DESCRIPTION
**Pull request by dtebbs**
- gtest
- removed some duplicated code
- build updates

### CODE REVIEW

File: libff/CMakeLists.txt [hunk 1]
```diff
@@ -24,13 +24,20 @@ file(
...
-  algebra/curves/bw6_761/*.cpp
+  algebra/curves/alt_bn128/*.?pp
+  algebra/curves/alt_bn128/*.tcc
+  algebra/curves/edwards/*.?pp
+  algebra/curves/edwards/*.tcc
+  algebra/curves/mnt/mnt4/*.?pp
+  algebra/curves/mnt/mnt4/*.tcc
+  algebra/curves/mnt/mnt46_common.?pp
+  algebra/curves/mnt/mnt46_common.tcc
+  algebra/curves/mnt/mnt6/*.?pp
```
Comment (AntoineRondelet) [inline libff/CMakeLists.txt:35]: Why did you add the hpp and tcc to the LIBFF_SOURCE variable?

Comment (dtebbs) [reply libff/CMakeLists.txt:35]: I was having some issues where things were not rebuilding correctly in incremental builds.  I think this adds more dependencies to the makefile (i.e. so it knows to rebuild, say, a test if some .tcc changes).

Comment (AntoineRondelet) [reply libff/CMakeLists.txt:35]: Got it.
Shall we do:

```
algebra/curves/**.?pp algebra/curves/**.tcc
```
etc. instead of listing the curves here (all supported curves are added here, so the shortcut notation above is equivalent)

File: libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp [hunk 2]
```diff
@@ -144,62 +144,10 @@ bool alt_bn128_G1::operator!=(const alt_bn128_G1& other) const
...
-    alt_bn128_Fq J = H * I;                              // J = H * I
-    alt_bn128_Fq r = S2_minus_S1 + S2_minus_S1;          // r = 2 * (S2-S1)
-    alt_bn128_Fq V = U1 * I;                             // V = U1 * I
-    alt_bn128_Fq X3 = r.squared() - J - (V+V);           // X3 = r^2 - J - 2 * V
-    alt_bn128_Fq S1_J = S1 * J;
-    alt_bn128_Fq Y3 = r * (V-X3) - (S1_J+S1_J);          // Y3 = r * (V-X3)-2 S1 J
-    alt_bn128_Fq Z3 = ((this->Z+other.Z).squared()-Z1Z1-Z2Z2) * H; // Z3 = ((Z1+Z2)^2-Z1Z1-Z2Z2) * H
-
-    return alt_bn128_G1(X3, Y3, Z3);
+    return add(other);
```
Comment (AntoineRondelet) [inline libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp:147]: Note, this can also be done for G2 and for other groups (i.e. other curves)

**Review by AntoineRondelet — COMMENTED** [review]
LGTM, modulo the few comments above.

File: libff/CMakeLists.txt (hunk 1 above)
Comment (dtebbs) [reply libff/CMakeLists.txt:35]: Ah, yes.  I didn't mention I originally tried this, but it brought in all the profile and test files.  It started to get awkward to correctly filter them all.  In a separate change we should probably consistently keep those non lib files in their own dirs.

Comment (AntoineRondelet) [reply libff/CMakeLists.txt:35]: Ah yes of course... I see. Alright let's stick to that then!

File: libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp (hunk 2 above)
Comment (dtebbs) [reply libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp]: I do the corresponding change for G2 in another PR, so I've removed this change from here for now.

Comment (AntoineRondelet) [reply libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp]: Ah great, thanks

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks @dtebbs
